Tests/DefaultXpringClientTest.swift

```diff
@@ -68,6 +68,124 @@ final class DefaultXpringClientTest: XCTestCase {
+    }
+
+    // THEN the engine result code is as expected.
+    XCTAssertEqual(transactionHash, TransactionHash.testTransactionHash)
+  }
+
+  func testSendWithClassicAddress() {
+    // GIVEN a classic address.
+    guard let classicAddressComponents = Utils.decode(xAddress: .destinationAddress) else {
+      XCTFail("Failed to decode X - Address.")
+      return
+    }
+    let xpringClient = DefaultXpringClient(networkClient: FakeNetworkClient.successfulFakeNetworkClient)
+
+    // WHEN XRP is sent to a classic address THEN an error is thrown.
+    XCTAssertThrowsError(try xpringClient.send(
+      .sendAmount,
+      to: classicAddressComponents.classicAddress,
+      from: .wallet
+      ))
+  }
+
+  func testSendWithInvalidAddress() {
+    // GIVEN a Xpring client and an invalid destination address.
+    let xpringClient = DefaultXpringClient(networkClient: FakeNetworkClient.successfulFakeNetworkClient)
+    let destinationAddress = "xrp"
+
+    // WHEN XRP is sent to an invalid address THEN an error is thrown.
+    XCTAssertThrowsError(try xpringClient.send(
+      .sendAmount,
+      to: destinationAddress,
+      from: .wallet
+      ))
+  }
+
+  func testSendWithAccountInfoFailure() {
+    // GIVEN a Xpring client which will fail to return account info.
+    let networkClient = FakeNetworkClient(
+      accountInfoResult: .failure(XpringKitTestError.mockFailure),
+      feeResult: .success(.testGetFeeResponse),
+      submitTransactionResult: .success(.testSubmitTransactionResponse),
+      transactionStatusResult: .success(.testGetTxResponse)
+    )
+    let xpringClient = DefaultXpringClient(networkClient: networkClient)
+
+    // WHEN a send is attempted then an error is thrown.
+    XCTAssertThrowsError(try xpringClient.send(
+      .sendAmount,
+      to: .destinationAddress,
+      from: .wallet
+      ))
+  }
+
+  func testSendWithFeeFailure() {
+    // GIVEN a Xpring client which will fail to return a fee.
+    let networkClient = FakeNetworkClient(
+      accountInfoResult: .success(.testGetAccountInfoResponse),
+      feeResult: .failure(XpringKitTestError.mockFailure),
+      submitTransactionResult: .success(.testSubmitTransactionResponse),
+      transactionStatusResult: .success(.testGetTxResponse)
+    )
+    let xpringClient = DefaultXpringClient(networkClient: networkClient)
+
+    // WHEN a send is attempted then an error is thrown.
+    XCTAssertThrowsError(try xpringClient.send(
+      .sendAmount,
+      to: .destinationAddress,
+      from: .wallet
+      ))
+  }
+
+  func testSendWithLatestLedgerSequenceFailure() {
+    // GIVEN a Xpring client which will fail to return the latest validated ledger sequence.
+    let networkClient = FakeNetworkClient(
+      accountInfoResult: .success(.testGetAccountInfoResponse),
+      feeResult: .failure(XpringKitTestError.mockFailure),
+      submitTransactionResult: .success(.testSubmitTransactionResponse),
+      transactionStatusResult: .success(.testGetTxResponse)
+    )
```

Comment:
Isn't this the same FakeNetworkClient config as the above test `testSendWithFeeFailure`? Looks like both have the feeResult field set to .failure with the rest as success.

Reply:
Ah, my fault. 

In the old protocol buffers, the fee / last ledger were different RPCs but now they're the same.